File: frontend/src/api/risk.ts
import axios from 'axios'

// F2.2 风险评估模块挂在 /api/risk/*（无 /v1 前缀）。
// http 实例 baseURL 为 /api/v1，直接复用会把 /api/risk/... 拼成 /api/v1/api/risk/...，
// 因此这里创建独立的实例：baseURL=/api + /risk/... = /api/risk/...
export const riskHttp = axios.create({ baseURL: '/api', timeout: 120000, headers: { 'Content-Type': 'application/json' } })
riskHttp.interceptors.request.use((config) => {
  config.headers['X-Trace-Id'] = crypto.randomUUID()
  const token = sessionStorage.getItem('access_token')
  if (token) config.headers.Authorization = `Bearer ${token}`
  return config
})
riskHttp.interceptors.response.use((response) => response, async (error) => {
  const original = error.config
  if (error.response?.status === 401 && original && !original._retry && sessionStorage.getItem('refresh_token')) {
    original._retry = true
    try {
      const { authApi } = await import('./auth')
      const r = await authApi.refresh(sessionStorage.getItem('refresh_token')!)
      sessionStorage.setItem('access_token', r.data.data.access_token)
      sessionStorage.setItem('refresh_token', r.data.data.refresh_token)
      window.dispatchEvent(new CustomEvent('auth:changed'))
      original.headers.Authorization = `Bearer ${r.data.data.access_token}`
      return riskHttp(original)
    } catch { /* fall through to error */ }
  }
  const detail = error.response?.data?.error?.message || error.response?.data?.detail || error.message || '请求失败'
  return Promise.reject(new Error(detail))
})

export interface RiskOption { key: string; text: string; score: number }
export interface RiskQuestion { q: number; dimension: string; question: string; options: RiskOption[] }
export interface RiskQuestionnaire { questionnaire_id: string; version: string; total_questions: number; dimensions: string[]; items: RiskQuestion[] }
export interface RiskAssessmentResult { customer_id: string; score: number; risk_level: string; level_name: string; answered: number; expired_at: string | null }
export interface SuitabilityCheckResult { customer_id: string; product_id: string; product_name: string | null; customer_risk_level: string; product_risk_level: string; matched: boolean; warning: string | null; max_allowed_product_risk: string | null }

export const riskApi = {
  questionnaire: () => riskHttp.get<{ data: RiskQuestionnaire }>('/risk/questionnaire'),
  submitAssessment: (customer_id: string, answers: { q: number; a: string }[]) =>
    riskHttp.post<{ data: RiskAssessmentResult }>('/risk/assessment', { customer_id, answers }),
  suitabilityCheck: (customer_id: string, product_id: string) =>
    riskHttp.post<{ data: SuitabilityCheckResult }>('/risk/suitability-check', { customer_id, product_id }),
}

File: app/schemas/risk.py
from datetime import datetime

from pydantic import BaseModel, Field


class RiskQuestionOption(BaseModel):
    key: str = Field(min_length=1, max_length=1)
    text: str
    score: int = Field(ge=0, le=100)


class RiskQuestionItem(BaseModel):
    q: int = Field(ge=1)
    dimension: str
    question: str
    options: list[RiskQuestionOption] = Field(min_length=4, max_length=4)


class RiskQuestionnaireResponse(BaseModel):
    questionnaire_id: str
    version: str
    total_questions: int
    dimensions: list[str] = Field(default_factory=list)
    items: list[RiskQuestionItem]


class RiskAnswer(BaseModel):
    q: int = Field(ge=1)
    a: str = Field(min_length=1, max_length=1)


class RiskAssessmentRequest(BaseModel):
    customer_id: int | str
    answers: list[RiskAnswer] = Field(min_length=16, max_length=16)


class RiskAssessmentResult(BaseModel):
    customer_id: int | str
    score: int = Field(ge=0, le=100)
    risk_level: str
    level_name: str
    answered: int
    expired_at: datetime | None = None


class SuitabilityCheckRequest(BaseModel):
    customer_id: int | str
    product_id: str = Field(min_length=1)


class SuitabilityCheckResult(BaseModel):
    customer_id: int | str
    product_id: str
    product_name: str | None = None
    customer_risk_level: str
    product_risk_level: str
    matched: bool
    warning: str | None = None
    max_allowed_product_risk: str | None = None


class RiskAlertHandleRequest(BaseModel):
    note: str = Field(default="", max_length=500)


class RiskMonitorRequest(BaseModel):
    """F4.1 交易事件接收：POST /api/risk/monitor 请求体。"""

    customer_id: int | str
    transaction_id: str = Field(default="", max_length=128)
    amount: float = Field(gt=0)
    transaction_type: str = Field(default="transfer", max_length=32)
    timestamp: datetime | None = None
